File: java/poker/src/main/java/com/duchen/poker/model/type/KingTypeChecker.java
package com.duchen.poker.model.type;

import java.util.Arrays;
import java.util.List;

public class KingTypeChecker implements TypeChecker {

    @Override
    public CardGroupType getType() {
        return CardGroupType.KING_CARD;
    }

    @Override
    public List<Integer> getMyPossibleCardCount() {
        return Arrays.asList(2);
    }

    @Override
    public int getMaxCardIfIsMyType(List<Integer> cards) {
        if (cards.get(0) >= 16 && cards.get(1) >= 16) {
            return 17;
        } else {
            return -1;
        }
    }
}

File: java/poker/src/main/kotlin/poker/model/type/KingTypeChecker.kt
package com.duchen.poker.model.type

import java.util.Arrays

class KingTypeChecker : TypeChecker {

    override val type: CardGroupType
        get() = CardGroupType.KING_CARD

    override val myPossibleCardCount: List<Int>
        get() = Arrays.asList(2)

    override fun getMaxCardIfIsMyType(cards: MutableList<Int>): Int {
        return if (cards[0] >= 16 && cards[1] >= 16) {
            17
        } else {
            -1
        }
    }
}
